theories/distributions/Distr.ec

```diff
@@ -1049,6 +1049,17 @@ move=> a ad @/(\o) @/pred1; apply/eq_iff; split; last exact: canLR.
 by move=> faE; move/can_gf: ad; rewrite faE.
 qed.
 
+lemma nosmt dmap1E_can_local (d: 'a distr) (f: 'a -> 'b) (g: 'b -> 'a) (x: 'b):
+  f (g x) = x =>
```

Comment:
Use the `cancel` predicate.

Reply:
Not using the cancel predicate is half the point. The cancel predicate states that g is the right inverse of f. The current statement only considers what happens at x, for which prior information might be available.

Reply:
You are right, I misread the quantification